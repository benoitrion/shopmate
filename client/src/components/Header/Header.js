import React from 'react';

import TopAppBar from '@material/react-top-app-bar';
import MaterialIcon from '@material/react-material-icon';

import '@material/react-top-app-bar/index.scss';
import '@material/react-material-icon/index.scss';

import './Header.scss';

function Header() {
  return (
    <React.Fragment>
      <TopAppBar 
        className='top-app-bar'
        title='SHOPMATE'
        navigationIcon={<MaterialIcon
          icon='menu'
          onClick={() => console.log('click')}
        />}
        actionItems={[
          <MaterialIcon icon='search' />,
          <MaterialIcon icon='shopping_basket' />,
        ]}
      />
    </React.Fragment>
  );
}

export default Header;
