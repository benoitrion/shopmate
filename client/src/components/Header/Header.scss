@import url('https://fonts.googleapis.com/css?family=Montserrat');

.top-app-bar {
  font-family: 'Montserrat', sans-serif;
  font-size: '24px';
  font-weight: 'bold';
  color: 'FFFFFF';
}