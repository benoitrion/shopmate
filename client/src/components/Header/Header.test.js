import React from 'react';
import { shallow } from 'enzyme';
import Header from './Header';
import TopAppBar from '@material/react-top-app-bar';

  describe('Header test', () => {
  let wrapper;
  beforeEach(() => wrapper = shallow(<Header />));

  it('should render correctly', () => expect(wrapper).toMatchSnapshot());
  
  it('renders without crashing', () => {
    expect(wrapper.find(TopAppBar).length).toEqual(1);
  });

});