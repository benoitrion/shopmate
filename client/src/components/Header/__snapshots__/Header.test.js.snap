// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header test should render correctly 1`] = `
<Fragment>
  <TopAppBar
    actionItems={
      Array [
        <MaterialIcon
          className=""
          hasRipple={false}
          icon="search"
          unbounded={false}
        />,
        <MaterialIcon
          className=""
          hasRipple={false}
          icon="shopping_basket"
          unbounded={false}
        />,
      ]
    }
    className="top-app-bar"
    dense={false}
    fixed={false}
    navigationIcon={
      <MaterialIcon
        className=""
        hasRipple={false}
        icon="menu"
        onClick={[Function]}
        unbounded={false}
      />
    }
    prominent={false}
    short={false}
    shortCollapsed={false}
    style={Object {}}
    tag="header"
    title="SHOPMATE"
  />
</Fragment>
`;
